src/spinegeneric/cli/check_data_consistency.py

```diff
@@ -68,12 +60,10 @@ def main():
         missing_subjects_folder.sort()
         pprint(missing_subjects_folder)
 
-    for dirName, subdirList, fileList in os.walk(data_path):
-        for file in fileList:
-            if file.endswith(".nii.gz"):
-                jsonSidecarPath = os.path.join(dirName, file.split(".")[0] + ".json")
-                if not os.path.exists(jsonSidecarPath):
-                    print("Missing jsonSidecar: " + jsonSidecarPath)
+    for img_path in args.path_in.glob("sub-*/**/*.nii.gz"):
+        json_path = img_path.with_suffix("").with_suffix(".json")
```

Comment:
Interesting! Chaining calls to `.with_suffix` to replace the suffix... :writing_hand: :thinking: